lib/print_my_barcode_job.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+# Wraps a print job for Print My Barcode
+# https://github.com/sanger/print_my_barcode
+class PrintMyBarcodeJob
+  PrintingError = Class.new(StandardError)
+
+  #
+  # Build a new print job
+  #
+  # @param printer_name [String] The hostname of the printer to use (eg. d304bc)
+  # @param label_template [String] The name of the label template to use as registered in Print my barcode
+  # @param labels [Array<Hash>] Array of hashes containing label content
+  #
+  def initialize(printer_name:, label_template:, labels:)
+    @printer_name = printer_name
+    @label_template = label_template
+    @labels = labels
+  end
+
+  #
+  # Send the print job to print my barcode to trigger printing
+  #
+  def save
+    v1? ? v1_request : v2_request
+  end
+
+  private
+
+  #
+  # Temporary method during transition. Determines if we are using the v1 or v2 API
+  # based on the uri
+  #
+  # @return [Bool] Returns true if we are using the v1 api
+  #
+  def v1?
+    pmb_uri.ends_with?('v1')
```

Comment:
It think that could be risky if we keep this temporary setting too long. Like for example if we add a '/' at the end of pmb_uri, this will fail but difficult to detect the problem. Maybe we can have something like having a separate v2? and v1? booleans that fully check for the url?